sdk/eventhub/event-hubs/src/eventDataBatch.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventData, toAmqpMessage } from "./eventData";
+import { ConnectionContext } from "./connectionContext";
+import { AmqpMessage } from "@azure/core-amqp";
+import { messageProperties, message } from "rhea-promise";
+
+/**
+ * A helper class for creating a batch of events; taking into account the max size limit, so that the `EventDataBatch`
+ * can be passed to the Send method of the `Sender` to send the `EventData` objects as a batch messages.
+ *
+ * Use the `tryAdd` function on the EventDataBatch to add events in a batch.
+ * @class
+ */
+export class EventDataBatch {
+  /**
+   * @property Describes the amqp connection context for the Client.
+   */
+  private _context: ConnectionContext;
+  /**
+   * @property Array of event data objects.
+   */
+  events: EventData[] = [];
+  /**
+   * @property A value that is hashed to produce a partition assignment.
+   * It guarantees that messages with the same partitionKey end up in the same partition.
+   * Specifying this will throw an error if the producer was created using a `paritionId`.
+   */
+  partitionKey?: string | undefined;
+  /**
+   * @property The maximum size allowed for the batch.
+   */
+  private readonly maxSize: number;
+  /**
+   * @property Current size of the batch in bytes.
+   */
+  currentSize: number;
+  /**
+   * @property Array of amqp message objects.
+   */
+  private messages: AmqpMessage[] = [];
```

Comment:
Its easy to be confused as to why we need this when we have the `events` property.
Either rename to `amqpMessages` or `encodedMessages` and update the description regarding the same